code/src/components/TodosList.js

```diff
@@ -0,0 +1,182 @@
+
+  let emptyState = ''
+
+  if (inputSearch === '') {
+    if (!allTodos.length) {
+      emptyState = <EmptyState
+        emptyIcon={faFaceGrinHearts}
+        emptyText={"Follow you mood and add some todo!"} />
+    } else if (pendingTodos.length && !completedTodos.length) {
+      emptyState = <EmptyState
+        emptyIcon={faFaceGrimace}
+        emptyText={"Complete some todos and feel the satisfaction!"} />
+    } else if (!pendingTodos.length && completedTodos.length) {
+      emptyState = <EmptyState
+        emptyIcon={faFaceGrinStars}
+        emptyText={"All todos are completed, hurrah!"} />
+    }
+  } else if (inputSearch !== '') {
+    if (searchTotal === 0 && allTodos.length) {
+      emptyState = <EmptyState
+        emptyIcon={faFaceMehBlank}
+        emptyText={"Nothing fits your search, maybe worth checking the spelling..."} />
+    } else if (!allTodos.length) {
+      emptyState = <EmptyState
+        emptyIcon={faFaceGrinHearts}
+        emptyText={"Follow you mood and add some todo!"} />
+    }
+  }
+
+  return (
+    <>
+      {allTodos.length ?
+        <Section>
+          <FilteringButtonsBox>
+            <FilteringButton
+              aria-label="display pending tasks"
+              disabled={isPendingDisabled}
+              onClick={displayPending}>
+              <FontAwesomeIcon icon={faCircle} />
+            </FilteringButton>
+            <FilteringButton
+              aria-label="display all tasks"
+              disabled={isAllDisabled}
+              onClick={displayAll}>
+              <FontAwesomeIcon icon={faInfinity} />
+            </FilteringButton>
+            <FilteringButton
+              aria-label="display completed tasks"
+              disabled={isCompletedDisabled}
+              onClick={displayCompleted}>
+              <FontAwesomeIcon icon={faCircleCheck} />
+            </FilteringButton>
+          </FilteringButtonsBox>
+          <FilteringInput>
+            <input
+              autoComplete="off"
+              onChange={onTodoSearch}
+              placeholder="Search a todo or category"
+              type="search"
+              value={inputSearch} />
+          </FilteringInput>
+          <Count>{count}</Count>
+        </Section>
+        :
+        ''
+      }
+      <section style={{ display: isPendingVisible }}>
+        <SortableDroppable arrayToUse={pendingTodosSearched.reverse().map((todo) => (
+          <Todo
+            id={todo.id}
+            key={todo.id}
+            setIsUndoDisabled={setIsUndoDisabled}
+            todo={todo} />
+        ))} />
+      </section>
+      <section style={{ display: isCompletedVisible }}>
+        <SortableDroppable arrayToUse={completedTodosSearched.reverse().map((todo) => (
+          <Todo
+            id={todo.id}
+            key={todo.id}
+            setIsUndoDisabled={setIsUndoDisabled}
+            todo={todo} />
+        ))} />
+      </section>
+      {emptyState}
+    </>
+  )
+}
+
+export default TodosList
```

Comment:
REALLY IMPRESIVE NADIA!! although very long/complex it is easy to understand and the logic is flawless.